docs/design/RUC-on-type.md

```diff
@@ -0,0 +1,397 @@
+# RequiresUnreferencedCode on Types
+`RequiresUnreferencedCode` is a feature for providing and suppressing diagnostics related to trimming. In general, when accessing a member attributed with `RequiresUnreferencedCode`, a warning should be produced at the access location that the member is not compatible with trimming. In addition, when a member is annotated with `RequiresUnreferencedCode`, trimming compatibility warnings will be suppressed in the body of that member.
+
+In the case in which `RequiresUnreferencedCode` is present on a class declaration, the following members behave as if they were annotated directly with the attribute:
+- Static methods
+- Constructors
+- Static properties
+- Static events
```

Comment:
I completely missed the events - we don't have any code to handle those, right? This will the first time we do any warnings on events (Data flow annotations don't apply to events, and you can't put RUC directly on event). But I think the solution should be very similar to fields.